providers/implementations/encode_decode/encode_key2text.c

```diff
@@ -272,6 +273,11 @@ static int dh_to_text(BIO *out, const void *key, int selection)
     if (params != NULL
         && !ffc_params_to_text(out, params))
         return 0;
+    length = DH_get_length(dh);
+    if (length > 0
+        && BIO_printf(out, "recommended-private-length: %ld bits\n",
+                      length) <= 0)
+        return 0;
```

Comment:
This looks like a new feature, but if you want to look at this like a bugfix, you probably also want the testsuite changes related to that.

Reply:
This is not a new feature. The recommended-private-length was printed in 1.1.1. Also there are testsuite changes already.

Reply:
#18480 has a "Checking recommended private key bits is correct" that's not in this PR.

Reply:
Because that tests the addition of the default key length to generated parameters which cannot be backported to 3.0.

Reply:
So to clarify, we can't test it (in the same way) because we don't have (an easy) way to actually set that value in the file.

Reply:
I've added a test in the evp_extra_test2.c that tests it for the known group. And cherry picked that commit to the master branch PR.